File: cs-orm/models/UserResponse.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace csorm.Models;

public class UserResponse
{
    public long Id { get; set; }
    public string Name { get; set; }
    public int Score { get; set; }

    public UserResponse(User user)
    {
        Id = user.Id;
        Name = user.Name ?? "Unknown";
        Score = CalculateScore(user);
    }

    private int CalculateScore(User u)
    {
        // Provide default values for null fields
        float paymentScore = (u.PaymentHistoryPercentOnTime ?? 0.0f) * 100f;
        float utilizationScore = Math.Max(0.0f, 1.0f - (u.CreditUtilizationRatio ?? 1.0f)) * 100f;
        float ageScore = Math.Min(u.CreditAgeYears ?? 0, 15) / 15f * 100f;
        float accountScore = Math.Min(u.TotalAccounts ?? 0, 10) / 10f * 100f;

        int inquiries = u.RecentInquiries ?? 0;
        int derog = u.DerogatoryMarks ?? 0;
        float inquiryPenalty = inquiries * 5 + derog * 10;
        float inquiryScore = Math.Max(0.0f, 100f - inquiryPenalty);

        float rawScore = paymentScore * 0.35f +
                         utilizationScore * 0.30f +
                         ageScore * 0.15f +
                         accountScore * 0.10f +
                         inquiryScore * 0.10f;

        float ficoScore = 300f + (rawScore / 100f) * 550f;
        return (int)Math.Min(Math.Round(ficoScore), 850);
    }
}

File: java-orm/src/main/java/com/java/orm/models/UserResponse.java
package com.java.orm.models;

import lombok.Data;

@Data
public class UserResponse {
    private long id;
    private String name;
    private int score;

    public UserResponse(User u) {
        this.id = u.getId();
        this.name = u.getName();
        this.score = calculateFicoScore(u);
    }

    private int calculateFicoScore(User u) {
        double paymentScore = u.paymentHistoryPercentOnTime * 100.0;
        double utilizationScore = Math.max(0.0, 1.0 - u.creditUtilizationRatio) * 100.0;
        double ageScore = Math.min(u.creditAgeYears, 15) / 15.0 * 100.0;
        double accountScore = Math.min(u.totalAccounts, 10) / 10.0 * 100.0;

        double inquiryPenalty = u.recentInquiries * 5.0 + u.derogatoryMarks * 10.0;
        double inquiryScore = Math.max(0.0, 100.0 - inquiryPenalty);

        double rawScore = paymentScore * 0.35 +
                utilizationScore * 0.30 +
                ageScore * 0.15 +
                accountScore * 0.10 +
                inquiryScore * 0.10;

        double fico = 300.0 + (rawScore / 100.0) * 550.0;
        return Math.min((int) Math.round(fico), 850);
    }
}
